app/backend/wells/views_v2.py

```diff
@@ -573,3 +577,170 @@ class WellDetail(WellDetailV1):
+            work_start_date=Case(
+                When(well_status__well_status_code=WELL_STATUS_CODE_CONSTRUCTION, then=F('construction_start_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_ALTERATION, then=F('alteration_start_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_DECOMMISSION, then=F('decommission_start_date')),
+                default=Value(None),
+                output_field=DateField()
+            ),
+            work_end_date=Case(
+                When(well_status__well_status_code=WELL_STATUS_CODE_CONSTRUCTION, then=F('construction_end_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_ALTERATION, then=F('alteration_end_date')),
+                When(well_status__well_status_code=WELL_STATUS_CODE_DECOMMISSION, then=F('decommission_end_date')),
+                default=Value(None),
+                output_field=DateField()
+            )
+        )
+
+        # Filtering for records missing any of the specified fields
+        missing_info_filter = (
+            Q(well_tag_number__isnull=True) |
+            Q(identification_plate_number__isnull=True) |
+            Q(well_class__isnull=True) |
+            Q(geom__isnull=True) | # for latitude and longitude
+            Q(finished_well_depth__isnull=True) |
+            Q(surface_seal_depth__isnull=True) |
+            Q(surface_seal_thickness__isnull=True) |
+            Q(aquifer_lithology__isnull=True) |
+            Q(well_status__isnull=True) |
+            Q(work_start_date__isnull=True) |
+            Q(work_end_date__isnull=True) |
+            Q(person_responsible__isnull=True) |
+            Q(company_of_person_responsible__isnull=True) |
+            Q(create_date__isnull=True) |
+            Q(create_user__isnull=True)
+            # Q(natural_resource_region__isnull=True) |
+            # Q(internal_comments__isnull=True)
+        )
+
+        queryset = queryset.filter(missing_info_filter)
+
+        # Additional filtering based on query parameters
+        work_start_date = self.request.query_params.get('work_start_date')
+        work_end_date = self.request.query_params.get('work_end_date')
+
+        if work_start_date:
+            queryset = queryset.filter(work_start_date__gte=work_start_date)
+        if work_end_date:
+            queryset = queryset.filter(work_end_date__lte=work_end_date)
+
+        return queryset
+    
+
+class CrossReferencingListView(ListAPIView):
+    serializer_class = CrossReferencingSerializer
+
+    swagger_schema = None
+    permission_classes = (WellsEditOrReadOnly,)
+    model = Well
+    pagination_class = APILimitOffsetPagination
+
+    # Allow searching on name fields, names of related companies, etc.
+    filter_backends = (WellListFilterBackend, BoundingBoxFilterBackend,
+                       filters.SearchFilter, WellListOrderingFilter, GeometryFilterBackend)
+    ordering = ('well_tag_number',)
+
+    def get_queryset(self):
```

Comment:
This queryset logic is reused in multiple locations, is it worth pulling this logic out into a common queryset util?

Reply:
moved to a utils method